### ISSUE TITLE
Training custom network throws error

### ISSUE THREAD

**Issue by TylerHomer** [tensorflow/agents#396]
I followed the recipes in the networks tutorial to create custom DDPG actor and critic networks, but upon reaching `agent.train(experience)`, the critic network throws `ValueError: Q-network should output a tensor of shape (batch,) but shape [64, 1] was returned.`

Full minimal example with trace [here](https://github.com/TylerHomer/tfagents-ddpg-test/blob/master/tf-agents-ddpg-example-custom-network.ipynb).

The custom critic network is this:
```
class CriticNetworkCustom(network.Network):

    def __init__(self,
                observation_spec,
                action_spec,
                name='CriticNetworkCustom'):
        # Invoke constructor of network.Network
        super(CriticNetworkCustom, self).__init__(
              input_tensor_spec=(observation_spec, action_spec), state_spec=(), name=name)

        self._obs_spec = observation_spec
        self._action_spec = action_spec

        # Encoding layer concatenates state and action inputs, adds dense layer:
        kernel_initializer = tf.keras.initializers.VarianceScaling(
            scale=1./3., mode='fan_in', distribution='uniform')
        combiner = tf.keras.layers.Concatenate(axis=-1)
        self._encoder = encoding_network.EncodingNetwork(
            (observation_spec, action_spec),
            fc_layer_params=(400,),
            preprocessing_combiner = combiner,
            activation_fn = tf.keras.activations.relu,
            kernel_initializer = kernel_initializer,
            batch_squash=True)

        # Initialize the custom tf layers here:
        self._dense1 = tf.keras.layers.Dense(400, name='Dense1')
        self._value_layer = tf.keras.layers.Dense(1,
                                                  activation=tf.keras.activations.linear,
                                                  name='Value') # Q-function output

    def call(self, observations, step_type=(), network_state=()):
        # Forward pass through the custom tf layers here (defined above):
        state, network_state = self._encoder(observations, 
                                             step_type=step_type, 
                                             network_state=network_state)
        state = self._dense1(state)
        value = self._value_layer(state)

        return value, network_state
```

**Comment by leroidauphin** [comment]
This looks like a really interesting problem, so I attempted to reproduce it to see if it could be fixed. I can't reproduce this in the latest release (0.5.0) with the same error message, but it is still not possible to train the agent. Comparing this custom critic network to the example critic networks in TF-Agents, it appears that the `value` tensor must indeed have shape `[batch_size,]`. 

The example critic networks just return this at the end of the `call` method: `tf.reshape(value, [-1])`. I tried that in a copy of your notebook and it seemed to work.
